src/libraries/Native/Unix/System.IO.Ports.Native/pal_serial.c

```diff
@@ -13,7 +13,7 @@
 #include <sys/socket.h>
 #include <pal_networking_common.h>
 
-// ENODATA is not defined in FreeBSD 10.3 but is defined in 11.0
+// ENODATA is not defined in FreeBSD
 #if defined(__FreeBSD__) & !defined(ENODATA)
 #define ENODATA ENOATTR
 #endif
```

Comment:
Same question here. And I'm saying that as the guy who originally added this code :smile:

Reply:
ENODATA doesn't appear to be defined in 12.2. I haven't checked if it was actually in 11.0.

Reply:
If it can't be there, not point checking for it. The build is complex enough without providing new means of indirection. Let past be the past and just remove the check :smile:

Reply:
OK, I need to look into this a bit more then. It might be better to put the condition around the usages as is done here for `ESOCKTNOSUPPORT` if we can never produce that error.

https://github.com/dotnet/runtime/blob/4d411739ce1d9db68dfe12227663479c9d71c50d/src/libraries/Native/Unix/Common/pal_error_common.h#L313-L324

Reply:
I've updated this as discussed. It builds fine but I'll try to run some unit tests as well.